cmd/internal/codegen/parse/crd.go

```diff
@@ -253,11 +265,28 @@ func (b *APIs) parseMapValidation(t *types.Type, found sets.String, comments []s
 var arrayTemplate = template.Must(template.New("array-template").Parse(
 	`v1beta1.JSONSchemaProps{
     Type:                 "array",
+    {{ if .MaxItems -}}
```

Comment:
does this work? these fields don't match the template underneath. is there some kind of unit test to test this?